packages/opentelemetry-exporter-collector/src/transform.ts

```diff
@@ -14,221 +14,195 @@
-  const spanTraceId = span.spanContext.traceId;
-
-  if (linkSpanId === spanParentId && linkTraceId === spanTraceId) {
-    return collectorTypes.LinkType.PARENT_LINKED_SPAN;
-  }
-  return collectorTypes.LinkType.UNSPECIFIED;
-}
-
-/**
- * converts span links
- * @param span
- */
-export function toCollectorLinks(span: ReadableSpan): collectorTypes.Links {
-  const collectorLinks: collectorTypes.Link[] = span.links.map((link: Link) => {
-    const collectorLink: collectorTypes.Link = {
-      traceId: hexToBase64(link.context.traceId),
-      spanId: hexToBase64(link.context.spanId),
-      type: toCollectorLinkType(span, link),
+  timedEvents: TimedEvent[]
+): opentelemetryProto.trace.v1.Span.Event[] {
+  const protoEvents: opentelemetryProto.trace.v1.Span.Event[] = [];
+  timedEvents.forEach(timedEvent => {
+    const timeUnixNano = core.hrTimeToNanoseconds(timedEvent.time);
+    const name = timedEvent.name;
+    const attributes = toCollectorAttributes(timedEvent.attributes || {});
+    const droppedAttributesCount = 0;
+
+    const protoEvent: opentelemetryProto.trace.v1.Span.Event = {
+      timeUnixNano,
+      name,
+      attributes,
+      droppedAttributesCount,
     };
 
-    if (link.attributes) {
-      collectorLink.attributes = toCollectorAttributes(link.attributes);
-    }
-
-    return collectorLink;
+    protoEvents.push(protoEvent);
   });
+  return protoEvents;
+}
 
-  return {
-    link: collectorLinks,
-    droppedLinksCount: 0,
-  };
+export function toCollectorLinks(
+  span: ReadableSpan
+): opentelemetryProto.trace.v1.Span.Link[] {
+  const protoLinks: opentelemetryProto.trace.v1.Span.Link[] = span.links.map(
+    (link: Link) => {
+      const protoLink: opentelemetryProto.trace.v1.Span.Link = {
+        traceId: core.hexToBase64(link.context.traceId),
+        spanId: core.hexToBase64(link.context.spanId),
+        attributes: toCollectorAttributes(link.attributes || {}),
+        droppedAttributesCount: 0,
+      };
+      return protoLink;
+    }
+  );
+  return protoLinks;
 }
 
-/**
- * @param span
- */
-export function toCollectorSpan(span: ReadableSpan): collectorTypes.Span {
+export function toCollectorSpan(
+  span: ReadableSpan
+): opentelemetryProto.trace.v1.Span {
   return {
-    traceId: hexToBase64(span.spanContext.traceId),
-    spanId: hexToBase64(span.spanContext.spanId),
+    traceId: core.hexToBase64(span.spanContext.traceId),
+    spanId: core.hexToBase64(span.spanContext.spanId),
     parentSpanId: span.parentSpanId
-      ? hexToBase64(span.parentSpanId)
+      ? core.hexToBase64(span.parentSpanId)
       : undefined,
-    tracestate: toCollectorTraceState(span.spanContext.traceState),
-    name: toCollectorTruncatableString(span.name),
-    kind: span.kind,
-    startTime: hrTimeToTimeStamp(span.startTime),
-    endTime: hrTimeToTimeStamp(span.endTime),
+    traceState: toCollectorTraceState(span.spanContext.traceState),
+    name: span.name,
+    kind: span.kind + 1,
```

Comment:
This is clever, but might break if we introduce new kind or if we (or proto) change the order of `SpanKind`. I think safer to handle with mapping, similar to [zipkin exporter](https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-exporter-zipkin/src/transform.ts#L23). WDYT?

Reply:
Definitely, want to avoid tricks that depend on the structure of the enum like that